packages/hadron-build/lib/download-center.js

```diff
@@ -1,198 +1,59 @@
-        return resolve(data);
-      }
-
-      if (err && err.code !== 'NotFound') {
-        return reject(err);
-      }
-      debug(`Uploading manifest for ${CONFIG.version}`);
-      req.upload({Body: JSON.stringify(MANIFEST, null, 2)}, (_err) => {
-        if (_err) return reject(_err);
-        resolve(MANIFEST);
-      });
-    });
-  });
-};
-
-let setup = () => {
+const getDownloadCenter = (bucketConfig) => {
   requireEnvironmentVariables([
     'DOWNLOAD_CENTER_AWS_ACCESS_KEY_ID',
     'DOWNLOAD_CENTER_AWS_SECRET_ACCESS_KEY'
   ]);
 
-  AWS.config.update({
+  return new DownloadCenter({
+    ...bucketConfig,
     accessKeyId: process.env.DOWNLOAD_CENTER_AWS_ACCESS_KEY_ID,
     secretAccessKey: process.env.DOWNLOAD_CENTER_AWS_SECRET_ACCESS_KEY
   });
 };
 
-exports.generateKey = (CONFIG) => {
-  CONFIG.download_center_id = CONFIG.id.split('-')[1];
-  CONFIG.download_center_key_prefix = CONFIG.download_center_id;
-  if (CONFIG.channel !== 'stable') {
-    CONFIG.download_center_key_prefix += `/${CONFIG.channel}`;
-  }
-  return CONFIG;
+const getKeyPrefix = (channel) => {
+  return channel && channel !== 'stable' ? `compass/${channel}` : 'compass';
 };
 
-exports.maybeUpload = (CONFIG) => {
-  setup();
-
-  if (CONFIG.channel === 'dev' || CONFIG.channel === 'alpha') {
-    debug('Skipping publish to S3 for %s channel.', CONFIG.channel);
-    return Promise.resolve();
-  }
-
-  exports.generateKey(CONFIG);
-
-  return Promise.all(CONFIG.assets.map( (asset) => maybeUpload(CONFIG, asset)))
-    .then(maybeUploadManifest(CONFIG));
+const uploadAsset = async(channel, asset) => {
+  const dlCenter = getDownloadCenter({ bucket: DOWNLOADS_BUCKET });
+  const objectKey = `${getKeyPrefix(channel)}/${asset.name}`;
+  return dlCenter.uploadAsset(objectKey, fs.createReadStream(asset.path));
 };
 
-exports.release = (id, version) => {
-  setup();
-
-  return new Promise((resolve, reject) => {
-    if (version.indexOf('-dev') > -1) {
-      debug('We dont post dev channel releases to the download center.');
-      return resolve(false);
-    }
-    let channel = version.indexOf('-beta') > -1 ? 'beta' : 'stable';
-    let dest = `com-download-center/${id}_${version.indexOf('-beta') > -1 ? '_beta' : ''}_latest.json`;
-    let src = `com-download-center/${id}/${version}.json`;
-    const s3 = new AWS.S3({
-      params: {
-        Bucket: MANIFEST_BUCKET,
-        Key: src
-      }
-    });
-
-    s3.headObject( (err, data) => {
-      if (data) {
-        debug(`copying ${src} -> ${dest}`);
-        return s3.copyObject({
-          Bucket: MANIFEST_BUCKET,
-          Key: dest,
-          CopySource: src
-        }, (_err, res) => {
-          if (_err) return reject(_err);
-
-          if (channel === 'stable') {
-            debug(`:dancers: ${id}@${version} is now available in the download center.`);
-          }
-          resolve(res);
-        });
-      }
+const downloadManifest = async(key = MANIFEST_OBJECT_KEY) => {
```

Comment:
Is this still used? Can we clean it up or do we plan to do something with it?

Reply:
Yes, this is used to diff manifests before update, print the diff for debugging purposes, warn if they are the same (which is an unexpected thing). See https://github.com/mongodb-js/compass/pull/3299#discussion_r933258183 and feel free to tell me whether it is okay to just warn in this case

Reply:
I found it, it was in a collapsed file